tests/test_util/test_util_fileutil.py

```diff
@@ -12,12 +12,13 @@
 from sphinx._cli.util.errors import strip_escape_sequences
 from sphinx.jinja2glue import BuiltinTemplateLoader
 from sphinx.util.fileutil import _template_basename, copy_asset, copy_asset_file
+from sphinx.util.template import BaseRenderer
 
 if TYPE_CHECKING:
     from sphinx.testing.util import SphinxTestApp
 
 
-class DummyTemplateLoader(BuiltinTemplateLoader):
+class DummyTemplateLoader(BuiltinTemplateLoader, BaseRenderer):
```

Comment:
Instances of this are given as the `renderer` parameter to `copy_asset_file` which is defined with `renderer: BaseRenderer | None = None`.